examples/bri-3/src/bri/utils/mapper.ts

```diff
@@ -0,0 +1,134 @@
+import { Injectable } from '@nestjs/common';
+import { Type, TypeKind } from 'tst-reflect';
+import { AccessModifier } from './types/access-modifiers';
+import Options from './types/options';
+
+/**
+ * Generic mapper to convert object to Type at runtime
+ *
+ * @author : Manik-Jain
+ */
+@Injectable()
+export default class Mapper {
+  /**
+   * type convert from provided input to provided Type
+   * provides capabilities to define some explicit
+   * keyerties on the type
+   *
+   * Be careful with providing options object as
+   * this will change the keyerties on the type
+   *
+   * @param from
+   * @param object
+   * @param options
+   * @returns generated object
+   */
+  map(from: any, object: Type, options?: Options) {
+    if (!from) throw new Error('undefined source');
+    const supportedPrefix = ['$', '_'];
+
+    const result = {};
+    const mismatchedKeys = [];
+    if (object.kind === TypeKind.Interface) {
+      const targetTypeKeys = object.getProperties().map((param) => param.name);
+      Object.keys(from).forEach((key) => {
+        if (targetTypeKeys.includes(key) && !options?.exclude?.includes(key)) {
+          result[key] = from[key];
+        } else {
+          mismatchedKeys.push(key);
+        }
+      });
+
+      if (options?.opts) {
+        const opts = options.opts;
+        Object.keys(opts).forEach((key) => {
+          if (
+            targetTypeKeys.includes(key) &&
+            !options?.exclude?.includes(key)
+          ) {
+            result[key] = opts[key];
+          } else {
+            mismatchedKeys.push(key);
+          }
+        });
+      }
+
+      if (process.env.DEBUG === 'true') {
+        console.log(`Type conversion successful to ${object.name}`);
+        if (mismatchedKeys.length > 0) {
+          console.log(
+            `Found ${mismatchedKeys.length} mismatched keys => ${mismatchedKeys}`,
+          );
+        }
+      }
+
+      return result;
+    } else if (object.kind === TypeKind.Class) {
+      const constructorParams = object.getConstructors()[0].getParameters();
+      const classProperties = object
+        .getProperties()
+        .filter((property) => property.accessModifier === 2)
```

Comment:
what is 2? Could it be moved to a constant at the top of the file or exchanged for an enum?